pkg/blockdeviceclaim/v1alpha1/build.go

```diff
@@ -323,3 +328,28 @@ func (b *Builder) WithBlockVolumeMode(mode corev1.PersistentVolumeMode) *Builder
 
 	return b
 }
+
+// WithBlockDevicePoolName appends (or creates) the BDC Label Selector
+// by setting the provided value to the fixed key
+// openebs.io/block-device-pool
+// This will enable the NDM to pick only devices that
+// match the node (hostname) and block device pool label.
+func (b *Builder) WithBlockDevicePoolName(bdPoolName string) *Builder {
```

Comment:
Should probably change this method name from `WithBlockDevicePoolName()` to something that is related to the field that is being populated by the builder. 
Consider `WithBlockDeviceSelector(selector)` / `WithSelectorLabels(key, value)` / `WithSelectorExpressions(key, operator, []value)`

Reply:
As this one is specifically setting the BlockDevicePoolName, it makes sense to keep this with current name. When generic labels are set they, we can use the generic names.